### PR TITLE
feat: added data/Bigint and N.between

### PR DESCRIPTION
**Pull request by jessekelly881**
Duplicated the functionality of data/Number to data/Bigint and added between and betweenExclusive filters for both modules. Using data/Bigint must be namespaced due to name collisions with number.

### CODE REVIEW

**Comment by gcanti** [comment]
Ok for Bigint.
Regarding "between", what if I want `-2 <= x < 2`? If we cannot find an API that allows for full configuration, I would suggest leaving "between" to be derived by end users by leveraging the existing combinators.

**Comment by jessekelly881** [comment]
Makes sense. That said, the @fp-ts/core/Number package exports a between function that is inclusive. The same argument could be made for this fn. What do you think if we copy that interface and remove the betweenExclusive fns but keep the between fns?

**Comment by gcanti** [comment]
> What do you think if we copy that interface and remove the betweenExclusive fns but keep the between fns? To keep things consistent between libraries

👍

**Comment by jessekelly881** [comment]
Updated

File: README.md [hunk 1]
```diff
@@ -483,12 +484,29 @@ pipe(S.number, S.negative()); // < 0
...
+### Bigint filters
+
+```ts
+import * as B from "@fp-ts/schema/data/Bigint";
+
+pipe(S.bigint, B.greaterThan(5n));
+pipe(S.bigint, B.greaterThanOrEqualTo(5n));
+pipe(S.bigint, B.lessThan(5n));
+pipe(S.bigint, B.lessThanOrEqualTo(5n));
+pipe(S.bigint, B.between(-2n, 2n)); // -2 <= x <= 2
```
Comment (gcanti) [inline README.md:496]: nitpick: `-2n <= x <= 2n` (same for the following comments)

Comment (jessekelly881) [reply README.md:496]: Ah good point

Comment (jessekelly881) [reply README.md]: I updated it

**Comment by gcanti** [comment]
Thank you @jessekelly881

**Pull request merged**